elastic/beats
# [Heartbeat] Force default gid access to extracted project files

## Summary

Fixes https://github.com/elastic/beats/issues/35343.

Add group access to temp directories and unzipped files generated for project monitors. Tested with push monitors and private locations.

Note: `local` and `zip_url` types are not fixed on this PR, as they will be removed in next release.


## Checklist




## How to test this PR locally



1. Set up a private location in Kibana.
2. Enroll an agent running as root.
3. Push browser monitors to run on private location.
4. Check generated temp directories and files include group access.
```
: docker exec -u root -it vigilant_wing ls -al /tmp
total 20
drwxrwxrwt 1 root root 4096 May 10 12:09 .
drwxr-xr-x 1 root root 4096 May 10 12:09 ..
drwxrwx--- 5 root root 4096 May 10 12:09 elastic-synthetics-unzip-2053533835
drwxrwx--- 5 root root 4096 May 10 12:09 elastic-synthetics-unzip-2310035686
drwxrwx--- 5 root root 4096 May 10 12:09 elastic-synthetics-unzip-652256972
```
